File: backend/services/admin/src/Enma.Admin.Application/Services/Super/SuperProjectsService.cs
using Enma.Admin.Application.Abstractions;
using Enma.Admin.Application.Contracts;
using Enma.Admin.Application.Dto.Super;
using Enma.Common.Models;
using FluentResults;

namespace Enma.Admin.Application.Services.Super;

internal sealed class SuperProjectsService : ISuperProjectsService
{
    private const int MaxPageSize = 200;

    private readonly IProjectsRepository _projectsRepository;

    public SuperProjectsService(IProjectsRepository projectsRepository)
    {
        _projectsRepository = projectsRepository;
    }

    public async Task<Result<PaginatedResult<SuperProjectListItemDto>>> ListAsync(
        int page,
        int pageSize,
        string? search,
        bool includeDeleted,
        Guid? organizationId,
        CancellationToken ct = default)
    {
        if (page < 1) page = 1;
        if (pageSize < 1) pageSize = 20;
        if (pageSize > MaxPageSize) pageSize = MaxPageSize;

        var listRes = await _projectsRepository.ListSuperAsync(page, pageSize, search, includeDeleted, organizationId, ct);
        if (listRes.IsFailed) return Result.Fail<PaginatedResult<SuperProjectListItemDto>>(listRes.Errors);

        var countRes = await _projectsRepository.CountSuperAsync(search, includeDeleted, organizationId, ct);
        if (countRes.IsFailed) return Result.Fail<PaginatedResult<SuperProjectListItemDto>>(countRes.Errors);

        return PaginatedResult<SuperProjectListItemDto>.Create(listRes.Value, countRes.Value, page, pageSize);
    }

    public async Task<Result<SuperProjectDetailsDto>> GetByIdAsync(Guid projectId, CancellationToken ct = default)
    {
        return await _projectsRepository.GetSuperDetailsAsync(projectId, ct);
    }
}

File: frontend/src/services/admin/super/superProjectsService.ts
import { apiClient } from "@/api/apiClient";
import type { Guid, PaginatedResult } from "@/types/admin.types";
import type {
  SuperProjectDetailsDto,
  SuperProjectListItemDto,
  SuperProjectsQuery,
} from "@/types/super-admin.types";

// Future: softDelete, restore, archive, unarchive.
export default class SuperProjectsService {
  private readonly baseUrl = "/api/admin/v1/super/projects";

  public async list(
    query: SuperProjectsQuery = {},
  ): Promise<PaginatedResult<SuperProjectListItemDto>> {
    const { data } = await apiClient.get<PaginatedResult<SuperProjectListItemDto>>(this.baseUrl, {
      params: {
        page: query.page ?? 1,
        pageSize: query.pageSize ?? 20,
        search: query.search || undefined,
        includeDeleted: query.includeDeleted ?? false,
        organizationId: query.organizationId || undefined,
      },
    });
    return data;
  }

  public async getById(projectId: Guid): Promise<SuperProjectDetailsDto> {
    const { data } = await apiClient.get<SuperProjectDetailsDto>(`${this.baseUrl}/${projectId}`);
    return data;
  }
}
